lang/rust/rust.talon

```diff
@@ -1,31 +1,42 @@
 tag: user.rust
 -
-tag(): user.code_imperative
-tag(): user.code_object_oriented
 tag(): user.code_comment_line
 tag(): user.code_comment_block_c_like
 tag(): user.code_comment_documentation
+
+tag(): user.code_imperative
+tag(): user.code_object_oriented
+
 tag(): user.code_data_bool
+tag(): user.code_data_null
+
 tag(): user.code_functions
 tag(): user.code_functions_gui
+
 tag(): user.code_libraries
 tag(): user.code_libraries_gui
+
 tag(): user.code_operators_array
 tag(): user.code_operators_assignment
 tag(): user.code_operators_bitwise
 tag(): user.code_operators_math
-tag(): user.code_operators_pointer
 
 settings():
     user.code_private_function_formatter = "SNAKE_CASE"
     user.code_protected_function_formatter = "SNAKE_CASE"
     user.code_public_function_formatter = "SNAKE_CASE"
     user.code_private_variable_formatter = "SNAKE_CASE"
+    user.code_protected_variable_formatter = "SNAKE_CASE"
     user.code_public_variable_formatter = "SNAKE_CASE"
 
+# rust-specific grammars
+
+## for unsafe rust
 state unsafe: "unsafe "
+unsafe block: user.code_state_unsafe()
```

Comment:
Any reason for making this its own command instead of just saying `state unsafe block` (i.e. chaining the `state unsafe` and `block` commands)?

Reply:
My thinking was because you can have an unsafe block, or an unsafe function. I thought relying on state unsafe would work for functions, but this would be quicker for blocks